src/renderers/WebGLRenderer.js

```diff
@@ -1922,10 +1922,14 @@ function WebGLRenderer( parameters = {} ) {
 
 				if ( _currentDrawBuffers.length !== 1 || _currentDrawBuffers[ 0 ] !== _gl.BACK ) {
 
-					_currentDrawBuffers[ 0 ] = _gl.BACK;
-					_currentDrawBuffers.length = 1;
+					if ( ! xr || ! xr.getSession() || ! xr.getSession().renderState || ! xr.getSession().renderState.layers ) {
 
-					needsUpdate = true;
+						_currentDrawBuffers[ 0 ] = _gl.BACK;
+						_currentDrawBuffers.length = 1;
+
+						needsUpdate = true;
+
+					} // else avoid https://github.com/mrdoob/three.js/issues/22079 when using XR layers.
```

Comment:
Can you explain in more detail why this fixes the mentioned warnings. TBH, I understand the change so far.

Reply:
I'm not sure of the exact flow. But I think the issue is that the system is trying to draw to `_gl.BACK` when using XR Layers rather than to the WebXR compositor's provided texture. I thought this would result in an error and a black frame since the draw calls would not be sent to the WebXR compositor's render target, but it only seems to be a warning.

Reply:
Sorry, I can't approve this PR without a closer study of the code.

Reply:
That's understandable. I restricted the code as much as possible so that it only affects devices with layers. At the moment, that's [limited to Oculus Browser](https://immersiveweb.dev/) so there is no other way to test this.

Another option might be to set `renderTarget` when in WebXR or when layers are active. That would cause the code to go through the `if ( renderTarget ) {` path above this change. I'll try making that change and see if the results are the same.

Reply:
The specific issue is that `drawBuffers([gl.BACK])` is only valid for the default framebuffer that you get from `bindFramebuffer(..., null)`. However, in WebXR mode, `state.bindFramebuffer(..., null)` has an override to replace the default framebuffer with the XR framebuffer which is a framebuffer object, and that should use `[gl.COLOR_ATTACHMENT0]` and not override with `[gl.BACK]`.

This is a general issue that happens when switching render targets while in XR, so it's not specific to WebXR Layers. It also happens when rendering shadows in phone AR mode.

See https://github.com/google/model-viewer/discussions/2769#discussioncomment-1307863 which has some more details from where I investigated this for model-viewer.

Adding a "is in XR" conditional here would work, but alternatively would it make sense to move the low-level `drawBuffers` call to WebGLState and apply an override there that replaces `gl.BACK` with `gl.COLOR_ATTACHMENT0` when xrFramebuffer is active?